roachtest: cdc/initial-scan-only/parquet failed

roachtest.cdc/initial-scan-only/parquet [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13388651?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13388651?buildTab=artifacts#/cdc/initial-scan-only/parquet) on master @ [f2917d82b235db6fda145a932050e5a58f957b9f](https://github.com/cockroachdb/cockroach/commits/f2917d82b235db6fda145a932050e5a58f957b9f):


```
(monitor.go:153).Wait: monitor failure: read tcp 172.17.0.3:34856 -> 35.231.106.116:29000: read: connection reset by peer
test artifacts and logs in: /artifacts/cdc/initial-scan-only/parquet/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=16</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-13388651/cdc-initial-scan-only-parquet/1704369809915/1704370435082)

</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*cdc/initial-scan-only/parquet.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35117

Getting many of these errors `logs/1.unredacted/cockroach-kv-distribution.teamcity-13444997-1704695584-43-n4cpu16-0001.ubuntu.2024-01-08T11_51_10Z.009099.log:I240108 11:51:10.966023 428 13@kv/kvserver/replicate_queue.go:785 ⋮ [T1,Vsystem,n1,replicate,s1,r1/1:‹/{Min-System/NodeL…}›] 3  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster` near the beginning of the test.

Looks like the test failed right at the beginning too

```
11:52:32 cdc.go:520: test status: creating cloudstorage changefeed into targets [tpcc.warehouse tpcc.district tpcc.customer tpcc.history tpcc.order tpcc.new_order tpcc.item tpcc.stock tpcc.order_line] with options (map[envelope:wrapped format:'parquet' initial_scan:'only' min_checkpoint_frequency:'10s'])
changefeed: 11:52:32 cdc.go:2643: Setting changefeed.default_range_distribution_strategy to balanced_simple
11:52:32 cdc.go:542: test status: created changefeed default with jobID 932722613759279105
11:56:55 cdc.go:345: test status: cdcTester closing
11:56:55 test_impl.go:414: test failure #1: full stack retained in failure_1.log: (monitor.go:153).Wait: monitor failure: read tcp 172.17.0.3:39266 -> 34.23.251.70:29000: read: connection reset by peer
11:56:55 test_runner.go:1126: test completed with failure(s)
11:56:55 test_runner.go:1160: skipping post test assertions as test failed
11:56:55 test_runner.go:1163: running test teardown (test-teardown.log)
11:57:24 test_runner.go:964: metrics: https://go.crdb.dev/roachtest-grafana/teamcity-13444997/cdc-initial-scan-only-parquet/1704714663150/1704715148589
11:57:24 test_runner.go:992: [w12] ##teamcity[testStarted name='cdc/initial-scan-only/parquet' flowId='cdc/initial-scan-only/parquet']
11:57:24 test_runner.go:1010: [w12] ##teamcity[testFailed name='cdc/initial-scan-only/parquet' details='(monitor.go:153).Wait: monitor failure: read tcp 172.17.0.3:39266 -> 34.23.251.70:29000: read: connection reset by peer|ntest artifacts and logs in: /artifacts/cdc/initial-scan-only/parquet/run_1' flowId='cdc/initial-scan-only/parquet']
11:57:24 test_runner.go:1014: [w12] --- FAIL: cdc/initial-scan-only/parquet (365.44s)
(monitor.go:153).Wait: monitor failure: read tcp 172.17.0.3:39266 -> 34.23.251.70:29000: read: connection reset by peer
test artifacts and logs in: /artifacts/cdc/initial-scan-only/parquet/run_1
11:57:25 issues.go:385: found existing GitHub issue: #117313
11:57:25 issues.go:439: created comment on existing GitHub issue (#117313)
11:57:25 test_runner.go:1024: [w12] ##teamcity[testFinished name='cdc/initial-scan-only/parquet' flowId='cdc/initial-scan-only/parquet' duration='365438']
```

--- a/pkg/ccl/changefeedccl/changefeed_processors.go
+++ b/pkg/ccl/changefeedccl/changefeed_processors.go
@@ -597,7 +597,12 @@ func (ca *changeAggregator) close() {
 		_ = ca.sink.Close()
 	}
 
-	ca.closeMetrics()
+	// The sliMetrics registry may hold on to some state for each aggregator
+	// (ex. last known resolved timestamp). De-register the aggregator so this
+	// data is deleted and not included in metrics.
+	if ca.sliMetrics != nil {
+		ca.sliMetrics.closeId(ca.sliMetricsID)
+	}
 
 	ca.memAcc.Close(ca.Ctx())
 
@@ -883,12 +888,6 @@ func (ca *changeAggregator) emitResolved(batch jobspb.ResolvedSpans) error {
 
 	ca.recentKVCount = 0
 	return nil
-}
-
-// closeMetrics de-registers the aggregator from the sliMetrics registry so that
-// it's no longer considered by the aggregator_progress gauge
-func (ca *changeAggregator) closeMetrics() {
-	ca.sliMetrics.closeId(ca.sliMetricsID)
 }
 
 // ConsumerClosed is part of the RowSource interface.

